cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -346,14 +374,396 @@ void AddSortingKernels(VectorKernel base, VectorFunction* func) {
   }
 }
 
+class TableSorter : public TypeVisitor {
+ private:
+  struct ResolvedSortKey {
+    ResolvedSortKey(const ChunkedArray& chunked_array, const SortOrder order)
+        : order(order) {
+      type = chunked_array.type().get();
+      null_count = chunked_array.null_count();
+      num_chunks = chunked_array.num_chunks();
+      for (const auto& chunk : chunked_array.chunks()) {
+        chunks.push_back(chunk.get());
+      }
+    }
+
+    template <typename ArrayType>
+    ArrayType* ResolveChunk(int64_t index, int64_t& chunk_index) const {
+      if (num_chunks == 1) {
+        chunk_index = index;
+        return static_cast<ArrayType*>(chunks[0]);
+      } else {
+        int64_t offset = 0;
+        for (size_t i = 0; i < num_chunks; ++i) {
+          if (index < offset + chunks[i]->length()) {
+            chunk_index = index - offset;
+            return static_cast<ArrayType*>(chunks[i]);
+          }
+          offset += chunks[i]->length();
+        }
+        return nullptr;
+      }
+    }
+
+    SortOrder order;
+    DataType* type;
+    int64_t null_count;
+    size_t num_chunks;
```

Comment:
`int64_t`?

Reply:
Fixed. It should be `int` because `ChunkedArray::num_chunks()` returns `int`.